cockroachcloud/egress-perimeter-controls.md

```diff
@@ -0,0 +1,418 @@
+
+1. Make a `POST` request ordering the API to update your cluster's egress policy from default allow-all to deny-all. This is needed before you can add egress rules to allowed external destinations.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    curl --request POST \
+    --header "Authorization: Bearer $CC_API_KEY" \
+    --header 'Cc-Version: 2022-09-20' \
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules/egress-traffic-policy" \
+    --data '{"allow_all":false}'
+    ~~~
+
+    ~~~txt
+    {}
+    ~~~
+
+## Create an egress rule to allow a destination
+
+An egress rule is created with the following attributes:
+
+- `name`: A name for the rule.
+- `type`: Whether the destination will be specified as a fully-qualified domain name (FQDN), or as a range of IP addresses specified in CIDR notation. Value is `"FQDN"` or `"CIDR"`.
+- `destination`: Either a fully qualified domain name, for example `www.cockroachlabs.com`, or a CIDR range, for example, `123.45.67.890/32`.
+- `ports`: An array of allowed ports, for example `[44,8080]`.
+    {{site.data.alerts.callout_danger}}
+    By default, all ports are allowed.
+    {{site.data.alerts.end}}
+- `paths`: An array of allowed destination paths, for example `[ "/docsrule/secretdocsdata/*", "/specialdata/superimportant/*" ]`.
+- `description`: The intended purpose of egress to the destination.
+
+The following steps create one FQDN rule and one CIDR rule.
+
+1. First, create a YAML manifest for a FQDN-based rule. This example adds egress to a couple of Google Cloud Platform (GCP) storage buckets.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~yaml
+    ---
+    name: "roach-buckets"
+    type: "FQDN"
+    destination: "storage.googleapis.com"
+    ports: [443, 80]
+    paths:
+        - "/customer-manged-bucket-1/*" # replace with bucket path
+        - "/customer-manged-bucket-2/*" # replace with bucket path
+    description: 'egress for GCP storage buckets'
+    ~~~
+
+1. Next, create a YAML manifest for a CIDR-based rule. This example adds egress to a self-managed Kafka cluster.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~yaml
+    --- # egress-rule2.yml
+    name: "roach-kafka"
+    type: "CIDR"
+    destination: "123.34.62.123/32" # replace with Kafka cluster CIDR range
+    ports: [443]
+    paths: []
+    description: 'egress for Kafka'    
+    ~~~
+
+1. Use Ruby (or another technique), to compile human-editable YAML into API-parsable JSON:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule1.yml > egress-rule1.json
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule2.yml > egress-rule2.json
+    ~~~
+
+1. Use the shell utility `jq` to inspect the JSON payload:
+    
+    {{site.data.alerts.callout_info}}
+    On macOS, you can install `jq` from Homebrew: `brew install jq`
```

Comment:
```suggestion
    On macOS, you can install `jq` from Homebrew: 
    
    {% include_cached copy-clipboard.html %}
    ~~~shell
    `brew install jq`
    ~~~
```
Very optional: I suspect this info is more usable if it's a code block, but I'll leave the decision entirely up to you.